argoverse/evaluation/stereo/eval.py

```diff
@@ -0,0 +1,367 @@
+    We consider the disparity of a pixel to be correctly estimated if the absolute disparity error is less than a
+    threshold and its relative error is less than 10% of its true value, similar to the KITTI Stereo 2015 benchmark [1].
+    We define three disparity error thresholds: 3, 5, and 10 pixels.
+
+    Some stereo matching methods such as Semi-Global Matching (SGM) might provide sparse disparity maps, meaning that
+    some pixels will not have valid disparity values. In those cases, we interpolate the predicted disparity map using
+    a simple nearest neighbor interpolation as in the KITTI Stereo 2015 benchmark [1] to assure we can compare it to
+    our semi-dense ground-truth disparity map. Current deep stereo matching methods normally predict disparity maps
+    with 100% density. Thus, an interpolation step is not needed for the evaluation.
+
+    The disparity errors metrics are the following:
+
+    1. all: Percentage of stereo disparity errors averaged over all ground truth pixels in the reference frame
+            (left stereo image).
+    2. bg: Percentage of stereo disparity errors averaged only over background regions.
+    3. fg: Percentage of stereo disparity errors averaged only over foreground regions.
+
+    The * (asterisk) means that the evaluation is performed using only the algorithm predicted disparities.
+    Even though the disparities might be sparse, they are not interpolated.
+
+    [1] http://www.cvlibs.net/datasets/kitti/eval_scene_flow.php?benchmark=stereo.
+
+Results:
+
+    We evaluate all metrics using three error thresholds: 3, 5, or 10 pixels.
+    The summary results are represented as a table containing the following fields:
+
+        all:10,
+        fg:10,
+        bg:10,
+        all*:10,
+        fg*:10,
+        bg*:10,
+        all:5,
+        fg:5,
+        bg:5,
+        all*:5,
+        fg*:5,
+        bg*:5,
+        all:3,
+        fg:3,
+        bg:3,
+        all*:3,
+        fg*:3,
+        bg*:3.
+
+    Note: The `evaluate` function will use all available logical cores on the machine.
+
+"""
+import argparse
+import logging
+import os
+from multiprocessing import Pool
+from pathlib import Path
+from typing import DefaultDict, List
+
+import cv2
+import disparity_interpolation
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+# absolute error thresholds in pixels, also used for KITTI stereo eval
+DEFAULT_ABS_ERROR_THRESHOLDS = [10, 5, 3]
+
+# relative error thresholds in pixels, also used for KITTI stereo eval
+DEFAULT_REL_ERROR_THRESHOLDS = [0.1, 0.1, 0.1]
+
+
+# The disparity error image uses a custom log-color scale depicting correct estimates in blue and wrong estimates in
+# red color tones, as in the KITTI Stereo 2015 Benchmark [1].
+# LOG_COLORMAP = [disparity error range, RGB color], where the disparity error range is defined as:
+# y = 2^x for x = [-Inf, -4, -3, -2, -1, 0, 1, 2, 3, 4, Inf].
+LOG_COLORMAP = [
```

Comment:
just out of curiosity, does the log colormap look a lot better than a linear colormap?

Reply:
Good question. I'm blindly using the KITTI's colormap, which looks pretty good already.